graylog2-server/src/main/java/org/graylog2/shared/rest/resources/ProxiedResource.java

```diff
@@ -32,4 +53,45 @@ public ProxiedResource(@Context HttpHeaders httpHeaders) {
             this.authenticationToken = null;
         }
     }
+
+    protected <T, K> Map<String, Optional<K>> getForAllNodes(Function<T, Call<K>> fn, Function<String, Optional<T>> interfaceProvider) {
+        return getForAllNodes(fn, interfaceProvider, Function.identity());
+    }
+
+    protected <T, K, L> Map<String, Optional<K>> getForAllNodes(Function<T, Call<L>> fn, Function<String, Optional<T>> interfaceProvider, Function<L, K> transformer) {
```

Comment:
It would be nice if you could use longer names for the types, T K L is a bit hard to handle while reading the code. :smile: